### DISCUSSION TITLE
[Solved/Workaround] Can I create an image from a font with the whole character map it contains?

### DISCUSSION THREAD

**Discussion by Feathered-Serpent** [ImageMagick/ImageMagick#3567] [Help] [answered]
Greetings everyone,

I know with the help of convert one can create an image with a specified text (as described at https://legacy.imagemagick.org/Usage/text/)

Is it also possible instead of inserting a text to create an image of the whole character map of a font file? Similar to the Windows Character Map?

With this one could make like a presentation image for a font file (like the name of the font at the top and beneath all glyphs), or might identify more easily, if there are more glyphs in one font file than in another of the same name.

**Comment by fmw42** [comment]
What version of Imagemagick and on what platform?  For Unix-like systems, see https://legacy.imagemagick.org/Usage/scripts/show_fonts

**Comment by snibgo** [comment, marked as answer]
You could make a text file that contained, say, the first 65535 characters in UTF-8, and render that in the fonts of your choice.

IM can't detect whether a font contains a glyph for a given code.

**Comment by Feathered-Serpent** [comment]
@fmw42 : I'm using Windows 10 x64

@snibgo : Thank you for the suggestion.. If ImageMagick doesn't see which glyphs are available, this might be the only solution. I shall mark this as answer.

**Comment by GeeMack** [comment]
I know this has been marked as solved, but I have a small BAT script that might be helpful to someone. This script runs on Windows 10, uses ImageMagick v7, and requires a Windows CLI version of "sed". I use the GnuWin32 version. The script takes one argument, the font name or the name of the actual font file.

    @echo off
    chcp 1252 > NUL
    setlocal
    
    if "%~1"=="" ( set IMFONT=Arial ) else ( set IMFONT="%~1" )
    echo Font: %IMFONT%
    
    echo. > chars.txt
    
    for %%A in ( 2 3 4 5 6 7 8 9 A B C D E F ) do (
       for %%B in ( 0 1 2 3 4 5 6 7 8 9 A B C D E F ) do (
          echo X | sed "s/.*/label:\x%%A%%B/g" >> chars.txt
       )
    )
    
    magick ^
       -respect-parenthesis ^
       -font %IMFONT% ^
       -background none ^
       -pointsize 30 ^
       -size 48x48 ^
       gradient:white-none ^
       -rotate 90 ^
       -set option:wide %%[w] ^
       -set option:wide16 %%[fx:w*16] ^
       -set option:high %%[h] ^
       ( -size %%[wide16]x%%[high] ^
          xc:none ^
          +font ^
          -gravity west ^
          -annotate +%%[fx:h*0.25]+0 %IMFONT% ^
          -gravity east ^
          +clone ^
          -composite ^
          -write mpr:header ^
          +delete ) ^
       -delete 0--1 ^
       -gravity center ^
       label:"\ " ^
       @chars.txt ^
       +gravity ^
       -duplicate %%[fx:16-(n%%16)],0 ^
       -background black ^
       -chop 1x1 ^
       -splice 1x1 ^
       +append ^
       +repage ^
       -crop %%[wide16]x0 ^
       -append ^
       +repage ^
       -gravity southeast ^
       -chop 1x1 ^
       -splice 1x1 ^
       -background white ^
       -flatten ^
       mpr:header ^
       +insert ^
       -append ^
       +repage ^
       -flatten ^
       -bordercolor black ^
       -shave 1 ^
       -border 1 ^
       -set filename:f char_%IMFONT% ^
          "%%[filename:f].png"
    
    exit /b

First it uses "sed" to generate a list of all the characters, each preceded by "label:", as a TXT file. Then the IM command reads that file and creates a label for each character. The bulk of the IM command just arranges the labels into a nice grid.

Say the script is named "chargrid.bat", and run with the command "chargrid.bat Times-New-Roman", this is the output image...

![char_Times-New-Roman](https://user-images.githubusercontent.com/65475279/115486654-bb4b7280-a21c-11eb-8d5e-272218b412d3.png)
